src/momento/responses/auth/generate_disposable_token.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import base64
+import json
+from abc import ABC
+
+from momento_wire_types import token_pb2 as token_pb
+
+from momento.responses.response import AuthResponse
+from momento.utilities.expiration import ExpiresAt
+
+from ..mixins import ErrorResponseMixin
+
+
+class GenerateDisposableTokenResponse(AuthResponse):
+    """Parent response type for a generate disposable token request.
+
+    The response object is resolved to a type-safe object of one of
+    the following subtypes:
+    - `GenerateDisposableToken.Success`
+    - `GenerateDisposableToken.Error`
+
+    Pattern matching can be used to operate on the appropriate subtype.
+    For example, in python 3.10+::
+
+        match response:
+            case GenerateDisposableToken.Success():
+                ...
+            case GenerateDisposableToken.Error():
+                ...
+            case _:
+                # Shouldn't happen
+
+    or equivalently in earlier versions of python::
+
+        if isinstance(response, GenerateDisposableToken.Success):
+            ...
+        elif isinstance(response, GenerateDisposableToken.Error):
+            ...
+        else:
+            # Shouldn't happen
+    """
+
+
+class GenerateDisposableToken(ABC):
+    """Groups all `GenerateDisposableTokenResponse` derived types under a common namespace."""
+
+    class Success(GenerateDisposableTokenResponse):
+        """Indicates the request was successful."""
+
+        auth_token: str
+        """The generated disposable token."""
+
+        endpoint: str
+        """The endpoint the Momento client should use when making requests."""
+
+        expires_at: ExpiresAt
+        """The time at which the disposable token will expire."""
+
+        def __init__(self, auth_token: str, endpoint: str, expires_at: ExpiresAt):
+            self.auth_token = auth_token
+            self.endpoint = endpoint
+            self.expires_at = expires_at
```

Comment:
I think you meant to make this a `dataclass`. Otherwise the variables you see here (`auth_token`, `endpoint`, `expires_at`) are _class variables_, not instance variables. They will be like static fields from other languages, which isn't what we want here. The `dataclass` decorator takes any class variable, turns it into an instance variable, and adds it to the generated `__init__`.